### PR TITLE
[Rebalancing] Compute target amounts utility function

### PR DESCRIPTION
**Pull request by uxd-vincent**
The goal of this PR is to introduce a new utility function that computes the target amounts of each depository based on the weights passed as parameter. This function will then be used by the rebalancing and super-mint/redeem set of IXs.

I am separating this function into a separate PR because:
- this will reduce the diff of future PR that actually introduce change in behavior in the smart contract
- this PR is extremely safe because it only introduces a utility function that is not yet used in the smart contract

### CODE REVIEW

File: programs/uxd/tests/unit_tests/utils/calculate_depositories_targets.rs [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+    }
+
+    #[test]
+    fn test_with_invalid_weight() -> Result<()> {
+        let circulating_supply = 1_000_000_000_000;
+
+        // Weights are invalid, does not add up to 100%
+        let identity_depository_weight_bps = percent_to_weight_bps(0);
+        let mercurial_vault_depository_0_weight_bps = percent_to_weight_bps(5);
+        let credix_lp_depository_0_weight_bps = percent_to_weight_bps(10);
```
[the remark this answers is not in the document]
Comment (uxd-vincent) [reply programs/uxd/tests/unit_tests/utils/calculate_depositories_targets.rs:193]: Yeah that sounds reasonable honestly 🤔

[the remark this answers is not in the document]
Comment (uxd-vincent) [reply programs/uxd/tests/unit_tests/utils/calculate_depositories_targets.rs]: done!

**Review by acamill — APPROVED** [review]
Lgtm. How does it fare in term of computing? (can log with log_compute_units or smthg, not needed just curious, if you know)

It's okay to fine tune later on if something is off as there is no hard bad effect to rebalance the money in the wrong depository, except credix where it's locked? Thanks for tiny pr

**Comment by uxd-vincent** [comment]
> Lgtm. How does it fare in term of computing? (can log with log_compute_units or smthg, not needed just curious, if you know)
> 
> It's okay to fine tune later on if something is off as there is no hard bad effect to rebalance the money in the wrong depository, except credix where it's locked? Thanks for tiny pr



> Lgtm. How does it fare in term of computing? (can log with log_compute_units or smthg, not needed just curious, if you know)
> 
> It's okay to fine tune later on if something is off as there is no hard bad effect to rebalance the money in the wrong depository, except credix where it's locked? Thanks for tiny pr

Thanks! Im not sure about the compute usage since this is not an IX yet, but it should be fairly low since we are only doing checked integer mathematics and nothing else. Will check in future PRs tho.

Also yeah I think we can adjust later if we want

**Pull request merged**